nengo/cache.py

```diff
@@ -109,20 +109,50 @@ class Fingerprint(object):
 
     __slots__ = ('fingerprint',)
 
-    SOLVERS = (Lstsq, LstsqL1, Nnls, NnlsL2, NnlsL2nz, LstsqNoise,
-               LstsqMultNoise, LstsqL2, LstsqL2nz, LstsqDrop)
-    LSTSQ_METHODS = (Cholesky, ConjgradScipy, LSMRScipy, Conjgrad,
-                     BlockConjgrad, SVD, RandomizedSVD)
-    NEURON_TYPES = (AdaptiveLIF, AdaptiveLIFRate, Direct, Izhikevich, LIF,
-                    LIFRate, RectifiedLinear, Sigmoid)
+    SOLVERS = (
+        Lstsq,
+        LstsqDrop,
+        LstsqL1,
+        LstsqL2,
+        LstsqL2nz,
+        LstsqNoise,
+        LstsqMultNoise,
+        Nnls,
+        NnlsL2,
+        NnlsL2nz,
+    )
+    LSTSQ_METHODS = (
+        BlockConjgrad,
+        Cholesky,
+        Conjgrad,
+        ConjgradScipy,
+        LSMRScipy,
+        RandomizedSVD,
+        SVD,
+    )
+    NEURON_TYPES = (
+        AdaptiveLIF,
+        AdaptiveLIFRate,
+        Direct,
+        Izhikevich,
+        LIF,
+        LIFRate,
+        RectifiedLinear,
+        Sigmoid
+    )
 
     WHITELIST = set(
         (bool, float, complex, bytes, list, tuple, np.ndarray) +
         int_types + string_types +
-        SOLVERS + LSTSQ_METHODS + NEURON_TYPES)
-    CHECKS = dict(
-        [(np.ndarray, check_dtype), (tuple, check_seq), (list, check_seq)] +
-        [(x, check_attrs) for x in SOLVERS + LSTSQ_METHODS + NEURON_TYPES])
+        SOLVERS + LSTSQ_METHODS + NEURON_TYPES
+    )
+    CHECKS = dict([
+        (np.ndarray, check_dtype),
+        (tuple, check_seq),
+        (list, check_seq)
+    ] + [
```

Comment:
There's no :barf: emoji but if there was I would use it on this line just gonna settle for :scream: